lib/MusicBrainz/Server/Controller/Area.pm

```diff
@@ -105,6 +110,37 @@ sub artists : Chained('load')
     $c->stash( artists => $artists );
 }
 
+=head2 events
+
+Shows all events for an area.
+
+=cut
+
+sub events : Chained('load')
+{
+    my ($self, $c) = @_;
+    my $events = $self->_load_paged($c, sub {
+        $c->model('Event')->find_by_area($c->stash->{area}->id, shift, shift);
+    });
+    $c->model('Event')->load_related_info(@$events);
+    $c->model('Area')->load(map { map { $_->{entity} } $_->all_places } @$events);
```

Comment:
The code works, but maps within maps are confusing (it can be hard to make sense of the brackets and which scope `$_` belongs to). This can be written more simply as:
```suggestion
    $c->model('Area')->load(map { $_->{entity} } map { $_->all_places } @$events);
```